utilities/vcf2json.py

```diff
@@ -295,56 +294,100 @@ def getMultADs(output_json, record, sample_position, alt, alt_ad_index, hasAD, n
 # Processes SnpEff and SnpSift output into final json file.
 
 
-def extractINFOField(output_json, record, codeDict):
+def extractINFOField(variant_id, patient_id, record, codeDict, mol_output, output_json):
     # Info field is dict with entries: ANN (SnpEff output) and POPAF (gnomAD output)
-
-    # Process the molecular consequence and predicted molecular impact
     if 'ANN' in record.INFO:
-        output_json["molecConseq"] = []
-        firstFlag = True
+        count = 3
         for ann in record.INFO['ANN']:
-            parseANN(output_json, ann, firstFlag, codeDict)
-            firstFlag = False
+            annList = ann.split('|')
+            isMane = False
 
-    if 'POPAF' in record.INFO:
-        for popAF in record.INFO['POPAF']:
-            if popAF is not None:
-                output_json["popAlleleFreq"] = float(popAF)
+            if annList[6] in transcript_map:
+                mane = transcript_map[annList[6]]
+                if(mane == 0 or mane == None):
+                    isMane = False
+                else:
+                    isMane = True
+
+            # Checking if MANE or not, If MANE is true for first MolecularConsequences then
+            # only one Value is added and others are ignored
+            # If MANE is true for 2nd MolecularConsequence add first and second only.
+            # If MANE is True for 3rd MolecularConsequences add first, second and third only.
+            # If no MANE Present only 2 MolecularConsequences are added.
+
+            if(count!=0):
```

Comment:
I'm struggling a bit to understand the logic here... `count` starts off as 3` on line 300. So if `isMane == True`, then you call `additionInMolecularConseq(...)` and `break`. Doesn't that `break` statement stop the `for` loop on line 301? If yes, then you'd only get the MANE transcript and that's it instead of two transcripts including the MANE one. I think the logic should look something like this:

```python
maneAdded = False
for pos, ann in enumerate(record.INFO['ANN']):
  annList = ann.split('|')

  # ... add the rest of the logic

  if pos == 0:
     additionInMolecularConseq(variant_id, patient_id, record, codeDict, mol_output, annList, isMane)
     maneAdded = isMane
  else if pos == 1 && isMane:
    additionInMolecularConseq(variant_id, patient_id, record, codeDict, mol_output, annList, isMane)
    maneAdded = True
  else if isMane:
    additionInMolecularConseq(variant_id, patient_id, record, codeDict, mol_output, annList, isMane)
    maneAdded = True

# Make sure we add the second transcript to the molecular consequences if we have more than one transcript and there is no MANE at all (not sure if this can happen, but let's make sure it's handled)
if not maneAdded and len(record.INFO['ANN']) > 1:
  annList = record.INFO['ANN'][1].split('|')
  additionInMolecularConseq(variant_id, patient_id, record, codeDict, mol_output, annList, isMane)
```

Disclaimer: I haven't tested this.